File: api/HomeBudget.API/HomeBudget.Application/DTOs/CategoriaDtos/UpdateCategoriaDto.cs
using HomeBudget.Domain.Entities;
using HomeBudget.Domain.Enums;

namespace HomeBudget.Application.DTOs.CategoriaDtos
{
    public class UpdateCategoriaDto
    {
        public Guid Id { get; set; }
        public required string Nome { get; set; }
        public required string Descricao { get; set; }
        public TipoCategoria Finalidade { get; set; }
        public DateTime DataModificacao { get; set; }
        public DateTime DataCriacao { get; set; }
        public List<Transacao> Transacoes { get; set; } = [];

    }
}

File: web/src/interfaces/CategoriaDtos/UpdateCategoriaDto.ts
import { TipoCategoria } from "../../enums/TipoCategoriaEnum";

export class UpdateCategoriaDto {
  id: string;
  nome: string = "";
  descricao: string = "";
  finalidade: number = TipoCategoria.Despesa;

  constructor(
    id: string = "",
    nome: string = "",
    descricao: string = "",
    finalidade: number = TipoCategoria.Despesa,
  ) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.finalidade = finalidade;
  }
}
